collector/collector.go

```diff
@@ -302,6 +302,13 @@ func (nc *NATSCollector) initMetricsFromServers(namespace string) {
 		}
 	}
 
+	if response == nil && retryTime > 0 {
```

Comment:
This will only retry once, which seems may solve the problem sometimes but in general I would think we should do it in a loop to wait until we've got some response.

I think it may be good to add an option like e.g. `--max-retries` to be able to bail out, but that can be done in a separate PR since it owuld also apply to other places where `opts.RetryInterval` is used.